synapse/lib/stormlib/aha.py

```diff
@@ -161,10 +258,210 @@ async def _del(self, svcname):
+            $_valu = $dict.$key
+            if ( $_valu != $lib.null ) {
+                if $_valu {
+                    return ( true )
+                } else {
+                    return ( false )
+                }
+            }
+            return ( $default )
+        }
+
+        $svc = $lib.aha.get($cmdopts.svc)
+        if ($svc = (null)) {
+            $lib.print(`No service found for: "{$cmdopts.svc}"`)
+        } else {
+            $services = $svc.services
+            if $services {
+                $lib.print(`Resolved {$cmdopts.svc} to an AHA Pool.\n`)
+                $lib.print(`The pool currently has {$lib.len($services)} members.`)
+
+                $lib.print(`AHA Pool:   {$svc.name}`)
+                for ($_svcname, $_svcinfo) in $services {
+                    $lib.print(`Member:     {$_svcname}`)
+                }
+            } else {
+                $lib.print(`Resolved {$cmdopts.svc} to an AHA Service.\n`)
+                $svcinfo = $svc.svcinfo
+                $leader = $svcinfo.leader
+                if ($leader = (null)) {
+                    $leader = 'Service did not register itself with a leader name.'
+                }
+                $lib.print(`Name:       {$svc.name}`)
+                $lib.print(`Online:     {$_getBoolKey($svcinfo, online, false)}`)
+                $lib.print(`Ready:      {$_getBoolKey($svcinfo, ready, null)}`)
+                $lib.print(`Run iden:   {$svcinfo.run}`)
+                $lib.print(`Cell iden:  {$svcinfo.iden}`)
+                $lib.print(`Leader:     {$leader}`)
+
+                if $cmdopts.nexus {
+                    if $svcinfo.online {
+                        $nexusOffset = $_getNexus($svc.name)
+                    } else {
+                        $nexusOffset = 'Service is not online. Will not attempt to retrieve its nexus offset.'
+                    }
+                    $lib.print(`Nexus:      {$nexusOffset}`)
+                }
+
+                $lib.print('Connection information:')
+                $urlinfo = $svcinfo.urlinfo
+                $keys = $lib.dict.keys($urlinfo)
+                $keys.sort()
+                for $k in $keys {
+                    $dk = `{$k}:`
+                    $dk = $dk.ljust(12)
+                    $lib.print(`    {$dk}{$urlinfo.$k}`)
+                }
+            }
+        }
+        '''
+    },
+    {
+        'name': 'aha.svc.list',
+        'descr': '''List AHA services.
+
+If the --nexus argument is given, the Cortex will attempt to connect to each service and report the Nexus offset of the service.
+
+The ready column indicates that a service has entered into the realtime change window for synchronizing changes from its leader.''',
+        'cmdargs': (
+            ('--nexus', {'help': 'Try to connect to online services and report their nexus offset.',
+                         'default': False, 'action': 'store_true'}),
+        ),
+        'storm': '''
+        function _getNexus(svcname) {
```

Comment:
Should we functionalize this as a `$lib.aha` method so it can be reused? ( return None if not online )

Reply:
I thought about that; but it felt a bit hyper-specific when we had some existing pieces to do this. We'd probably have to do a `isok, infodict` type API; if we wanted to message back any information about why we couldn't get the nexus information.